src/library/features/crates/cratefeaturehelper.cpp

```diff
@@ -56,7 +56,11 @@ CrateId CrateFeatureHelper::createEmptyCrate() {
             continue;
         }
 
-        if (m_pTrackCollection->crates().readCrateByName(newCrate.getName())) {
+        // select name from crateClosure join crates on id = childId group by childId having count(*) = 1;
+        // check if it has parent, if not compare with names from above
+        // in closure.
+        // Else split path into tokens and compare to those
+        if (m_pCrates->hierarchy().collectRootCrateNames().contains(newCrate.getName())) {
```

Comment:
This logic should be moved to CrateHierarchy. You could make a new function called something like `hasRootCrateNamed(const QString& crateName)`.